backbone/src/main/java/org/researchstack/backbone/step/Step.java

```diff
@@ -104,6 +105,26 @@ public void setOptional(boolean optional) {
         this.optional = optional;
     }
 
+    /**
+     * A boolean value indicating whether the flow need to stop after this step is completed.
+     * <p>
+     *
+     * @return a boolean indicating whether the step is the final step of the flow
+     */
+    public boolean isCompletion() {
```

Comment:
Opinion: would it be more clear if you call this `isFinalStepOfTheFlow()` and `setAsFinalStepOfTheFlow(boolean)` down below? I mean, the word `completion` to me, doesn't indicate much in this context. What does it mean to be "a completion = true"? 

Feel free to ignore or use something different, as this is my personal opinion. :)

Reply:
Completion step is not necessarily the final step of the flow (i.e. branching scenario)
I can rename it to isCompletionStep maybe?